File: rb101/rb101-rb109-small_problems/easy9/greetings.rb
def greetings(name_array, status_hash)
  name = name_array.join(' ')
  title = status_hash[:title]
  occupation = status_hash[:occupation]
  puts "Hello, #{name}! Nice to have a #{title} #{occupation} around."
end

greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' })

# ALTERNATIVE SOLUTION:

# def greetings(name, status)
#   "Hello, #{name.join(' ')}! Nice to have a #{status[:title]} #{status[:occupation]} around."
# end

# FURTHER EXPLORATION:

# See original solution above.

File: js210/lesson_5/greetings.js
function greetings(name, status) {
  let fullName = name.join(' ');
  let fullStatus = `${status.title} ${status.occupation}`

  console.log(`Hello ${fullName}! Nice to have a ${fullStatus} around.`);
}

greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' });

// console output
// Hello, John Q Doe! Nice to have a Master Plumber around.


// ALTERNATIVE SOLUTION:

// function greetings(name, status) {
//   let nameGreet = 'Hello, ' + name.join(' ') + '!';
//   let statusGreet = 'Nice to have a ' + status.title + ' ' + status.occupation + ' around.';

//   console.log(nameGreet + ' ' + statusGreet);
// }
